Add documentation for the deployment of Node.js Function Apps (#511)
pagopa/dx: apps/website/docs/pipelines/node-function-app-deploy.md

---
sidebar_position: 1

> sidebar_label: Deployment of a Node.js Function App

Reviewer: ```suggestion
sidebar_label: Deploying a Node.js Application to Azure Function App
```

---

# Deployment of a Node.js Function App

The
[Node.js FunctionApp Deployment](https://github.com/pagopa/dx/blob/main/.github/workflows/function_app_deploy.yaml)
is a reusable GitHub Actions workflow designed for building and deploying
**Node.js** applications from a monorepo based on Turbo to Azure Function Apps.

## How It Works

> The workflow consists of two main jobs: `build` and `deploy`.

Reviewer 2: ```suggestion
The workflow consists of two main jobs:
```

> **Build Job (`build`)**:

Reviewer: same as https://github.com/pagopa/dx/pull/512/files#r2055527329

- Deploys the zip artifact:
  - If `use_staging_slot` is `true` (default), deploys to the `staging` slot.
  - If `use_staging_slot` is `false`, deploys directly to the `production` slot.
- If deployed to the `staging` slot, performs a slot swap to move the new
  version to `production`.

## Usage

To use this workflow, invoke it from your own GitHub Actions workflow file.

```yaml
name: Deploy My Function App

on:
  push:
    branches:
      - main

jobs:
  deploy_app:
    uses: pagopa/dx/.github/workflows/function_app_deploy.yaml@main
    name: Deploy Function App to Production
    with:
      workspace_name: "my-function-app-workspace" # The name of your app's workspace in turbo.json
      environment: "prod" # The target environment (e.g., dev, uat, prod)
      resource_group_name: "my-resource-group"
      function_app_name: "my-function-app"
      # use_staging_slot: true # Optional: default is true
      # use_private_agent: true # Optional: default is true
      # use_labels: false # Optional: default is false
      # override_labels: '' # Optional: default is ""
```

## Notes

- This workflow assumes your project is structured as a Turborepo monorepo.

> - The workspace being deployed must have a build script in its package.json.

Reviewer 2: ```suggestion
- The workspace being deployed must have a build script in its `package.json`.
```
